pkg/skaffold/render/renderer/util/util.go

```diff
@@ -45,7 +45,9 @@ type GenerateHydratedManifestsOptions struct {
 func GenerateHydratedManifests(ctx context.Context, out io.Writer, builds []graph.Artifact, g generate.Generator, labels map[string]string, ns string, opts GenerateHydratedManifestsOptions) (manifest.ManifestList, error) {
 	// Generate manifests.
 	rCtx, endTrace := instrumentation.StartTrace(ctx, "Render_generateManifest")
-	manifests, err := g.Generate(rCtx, out)
+	// TODO(https://github.com/GoogleContainerTools/skaffold/issues/8318): revisit this code to better tackle this case for kustomize.
+	// We could have a different renderer for kustomize that doesn't inject namespaces.
+	kustomizeManifests, otherManifests, err := g.Generate(rCtx, out)
```

Comment:
NOTE: @ericzzzzzzz has added a seperate `kustomize` renderer in his PR here:
https://github.com/GoogleContainerTools/skaffold/pull/8365

Reply:
Yes! 😄 with it we can tackle this case better.